packages/paste-core/components/pagination/src/PaginationNumbers.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {PaginationLabel} from './PaginationLabel';
+import type {PaginationNumbersProps} from './types';
+import {PaginationNumbersPropTypes} from './proptypes';
+
+const PaginationNumbers = React.forwardRef<HTMLUListElement, PaginationNumbersProps>(
+  ({children, pageLabel, ...props}, ref) => {
+    const [childrenCount, validChildren] = React.useMemo(
+      () => [
+        React.Children.count(children),
+        React.Children.toArray(children).filter((child) => React.isValidElement(child) || typeof child === 'string'),
+      ],
+      [children]
+    );
+    const keySeed = useUIDSeed();
+
+    return (
+      <>
+        <Box {...safelySpreadBoxProps(props)} ref={ref} as="ul" display="inline-flex" margin="space0" padding="space0">
+          {validChildren.map((child, index) => {
+            return (
+              <Box
+                key={keySeed(`pagination-numbers-${index}`)}
+                as="li"
+                listStyleType="none"
+                margin="space0"
+                marginRight={childrenCount !== index + 1 ? 'space30' : 'space0'}
+                padding="space0"
+              >
+                {child}
+              </Box>
```

Comment:
Looks like the ul and li styling of PaginationNumbers and PaginationItems is shared. Maybe abstract them rather than dupe across files?